sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts

```diff
@@ -500,4 +508,29 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
       transactionalEntityManager,
     );
   }
+
+  /**
+   * Get COE approval period status which defines
+   * if the COE can be confirmed by institution on current date.
+   * @param disbursementDate disbursement date.
+   * @param studyEndDate study end date of the offering.
+   * @returns COE approval period status.
+   */
+  getCOEApprovalPeriodStatus(
+    disbursementDate: string | Date,
+    studyEndDate: string | Date,
+  ): COEApprovalPeriodStatus {
+    if (!isBeforeGivenDaysFromNow(disbursementDate, COE_WINDOW)) {
```

Comment:
I got the idea of the new methods but we also have the `isBetweenPeriod` that is exactly to check if a date is between two other dates and also it is pretty explicit. Any particular reason why we did not use the existing method?

Reply:
To combine both the validations, the idea of `isBetweenPeriod` seems to be perfect. But because we are trying to read each type of warning, period couldn't be exactly used here.